lib/spotify/types.ts

```diff
@@ -25,3 +25,25 @@ export function isUserInfoResponse(res: any): res is UserInfoResponse {
     'uri' in res
   );
 }
+
+type TopSongs = {
+  'name': string,
+  'href': string, 
+  'uri': string, 
+  'popularity': number, 
+  'artists': Array<string>,
+};
```

Comment:
The type name doesn't make sense.. TopSongs implies multiple songs but this type reflects only a single song.

All database entries must have an `id`, so we should get the song id here too.

Remove all the quotes

Turn into an interface! An interface is extendable, whereas a type is static.

For example:

```ts
export interface IDbItem {
  id: string;
}

export interface ISong extends IDbItem {
  name: string;
}

/**
ISong is in the following form:

{
  id: string,
  name: string,
}
*/
```

Reply:
i think this should be in a separate file.. u need to create a song object at some point too!